python/ray/experimental/queue.py

```diff
@@ -0,0 +1,131 @@
+from collections import deque
+import ray
+import time
+
+
+class Empty(Exception):
+    pass
+
+
+class Full(Exception):
+    pass
+
+
+class Queue:
+    """Queue implementation on Ray.
+
+    Todo:
+        - not_empty
+        - not_full
+        - join
+        - mutex
+        - task_done
+        - unfinished_taskss
+    """
+    def __init__(self, maxsize=0):
+        self.maxsize = maxsize
+        self.actor = _QueueActor.remote(maxsize)
+
+    def qsize(self):
+        return ray.get(self.actor.qsize.remote())
+
+    def empty(self):
+        return ray.get(self.actor.qsize.remote())
+
+    def full(self):
+        return ray.get(self.actor.full.remote())
+
+    def put(self, item, block=True, timeout=None):
+        if self.maxsize <= 0:
+            self.actor.put.remote(item)
+        elif not block:
+            if not ray.get(self.actor.put.remote(item)):
+                raise Full
+        elif timeout is None:
+            # Polling
+            # TODO: implement with not_full condition variable
+            while not ray.get(self.actor.put.remote(item)):
```

Comment:
Currently blocks by polling which isn't the most elegant solution, this could have better performance with a `time.sleep` statement on each loop iteration or by creating some sort of condition variable abstraction.